string name
string property
int32 value
